x-pack/plugins/maps/public/classes/layers/layer.tsx

```diff
@@ -166,6 +170,14 @@ export class AbstractLayer implements ILayer {
     return this._descriptor;
   }
 
+  queryForTileMeta(mbMap: MbMap): Feature[] | null {
+    return null;
+  }
+
+  getMetaFromTiles(): Feature[] {
+    return this._descriptor.__metaFromTiles;
```

Comment:
__metaFromTiles can be undefined. How about returning an empty array if __metaFromTiles is not provided.